### PR TITLE
Document the .replace method

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes https://github.com/tskit-dev/tskit/issues/1503

### CODE REVIEW

File: docs/python-api.rst [hunk 1]
```diff
@@ -173,16 +173,19 @@ Tables
...
-of the data. For example::
+Each table supports accessing the data either by row or column.
+
+To access the data in a *column*, we can use standard attribute access which will
+return a copy of the column data as a numpy array. To access the data in a *row*, say
+row number ``j`` in table ``t``, simply use ``t[j]``. The returned row has attributes
+allowing contents to be accessed by name, e.g. ``site_table[0].position``,
+``sites_table[0].ancestral_state``, ``sites_table[0].metadata`` etc. Note that row
+contents cannot be changed. To create a new row with changed content,
+row objects have a ``replace`` method. For example::
```
Comment (jeromekelleher) [inline docs/python-api.rst:188]: Nice to link to a method, like "(e.g., :meth:`Site.replace`)" or something?

**Review by jeromekelleher — APPROVED** [review]
Looks good, minor suggestion

File: docs/python-api.rst (hunk 1 above)
Comment (hyanwong) [reply docs/python-api.rst:188]: Yeah, We don't list the actual method anywhere. I think either we'd need to list all the TableRow definitions, or perhaps just do:

```
.. autoclass:: Tskit.util.Dataclass
```

I did that, but the removed it, as then we'd be actually documenting this data class, and presumably therefore be committed to retaining it (and not e.g. implementing `replace` in a different way. So I thought it easiest not to commit to a particular implementation for the time being.

Comment (jeromekelleher) [reply docs/python-api.rst:188]: OK, let's leave it then.

**Pull request merged**